### ISSUE TITLE
[Feature] Add more info fields to studios

### ISSUE THREAD

**Issue by echo6ix** [stashapp/stash-box#676]
**Is your feature request related to a problem? Please describe.**
Other database sites do not seem to have much fundamental data about studios. Some additional fundamental fields would be helpful for archival/historical and prospective filtering reasons.

**Describe the solution you'd like**
To add the following core informational fields to studios:

FIELD NAME|DATA TYPE|DESCRIPTION|REQUIRED
--|--|--|--|
Aliases|multivalue strings|Studio aliases, such as official or popular acronyms|False
Start date|date|Studio launch date|False
End date|date|Studio end date|False
Niche|tags|Tags describing the primary niche of the studio (have a low limit of how many tags can be used here)|True
Orientation|tags|Sexual orientation category of studio content. Tag choices available to user `straight`, `gay`, `trans`, `bi`|True

This type of information can be curated from various sources including sites like https://www.thebestporn.com/review/nudeinfrance/ which document `start date`, `end date`, `primary niche`, and `secondary niche` values.

**Orientation**
This info would be useful in filtering StashDB content (excluding edits). Not unlike many major adult tube site platforms which prompt users for the type of sexual orientation category they prefer to see (gay, straight, trans).

## Edit
See comment https://github.com/stashapp/stash-box/issues/676#issuecomment-2397387341

> The same thing is happening with expanded studio fields that happened with performer and scene fields. Over time people begin thinking of good ideas for pragmatic fields. Or fields that are specific to StashDB, but may be unnecessary to other stash-boxes or vice versa.

> I'd propose making a system where the stash-box administrator can add custom fields to objects (studios, scenes, performers, tags, etc), rather than hard-coding them. After all, stash-box is more generalized than StashDB, as such offering custom fields makes it more adaptable.

**Comment by echo6ix** [comment]
We could also consider adding a field that denotes the `Production Type` of content

FIELD NAME|DATA TYPE|DESCRIPTION|REQUIRED
---|---|---|---
Production Type|String options|`Original` This option indicates studios that exclusively produce their own original content.<br />`Redistribution` This option indicates studios that redistribute content only|False

This would primarily help with identifying en masse studios that just redistribute content. However, this field could also be considered redundant as the `Niche` field could also be used to denote this.

**Comment by InfiniteStash** [comment]
@echo6ix Regarding orientation, we should probably add lesbian as well to differentiate MM/FF. Also, is the idea that it should be multiple choice?

**Comment by echo6ix** [comment]
> @echo6ix Regarding orientation, we should probably add lesbian as well to differentiate MM/FF.

Correct, I agree.

> Also, is the idea that it should be multiple choice?

That was my idea. Do you think multiple choice sexual orientation would be suitable for 99% of studios?

**Comment by DogmaDragon** [comment]
I don't think making any of the suggested fields mandatory is needed. In fact it would be a net negaive in some other stash-box applications that are not focused on more structured mainstream content.

**Comment by echo6ix** [comment]
@DogmaDragon Fair point. I did not think of that. And in retrospect, even on StashDB, if there are parent studios that cover a broad span of sexual orientations and niches, those "required" fields would not be applicable.

**Comment by Ronnie711** [comment]
> > That was my idea. Do you think multiple choice sexual orientation would be suitable for 99% of studios?

AdultTime releases across all orientations interchangably within most of their series.

**Comment by echo6ix** [comment]
> AdultTime releases across all orientations interchangably within most of their series.

Hmm. I am browsing https://www.adulttime.com/en/channels. They have four sexual orientation tags, `lesbian`, `bisexual`, `gay` and `trans`. I am not seeing any studios with multiple sexual orientation tags on that list. Scrolling through it, the studio niches seem like they can be segregated by orientation. Is there another list with different studio categories?

**Comment by Ronnie711** [comment]
> > AdultTime releases across all orientations interchangably within most of their series.
> 
> Hmm. I am browsing https://www.adulttime.com/en/channels. They have four sexual orientation tags, `lesbian`, `bisexual`, `gay` and `trans`. I am not seeing any studios with multiple sexual orientation tags on that list. Scrolling through it, the studio niches seem like they can be segregated by orientation. Is there another list with different studio categories?

Couple Swapping: https://stashdb.org/studios/c19e9f33-fb6a-4f20-9980-ad8cd6761f87 < - Both Gay & straight content

Caught Fapping: https://stashdb.org/studios/48c86cd1-4a11-43ab-a303-5da49539266f <- Gay, Straight & Trans scenes

Two quick examples, I have no interest in Trans/Gay content so have become acustomed to double checking their content before I 'collect' entire series.

**Comment by echo6ix** [comment]
Thank you for the links. I'm a total lightweight here. I guess this would be an ineffective way to filter all studios by orientation. But still totally suitable as a way to tag studios by their primary categories (and sexual orientation tags where it is applicable).

**Comment by Borisbee1** [comment]
Chiming in here to say that I think the alias field would be very useful especially when dealing with jav studios that can often have slightly different names depending on where you're scraping from or how it's translated.

**Comment by Rival98-stash** [comment]
The studio alias will help with the maintenance of the Community scrapers. There have been many examples of the scrapers not matching the studio name. This could be because:

- Changing html markup on the video webpage 
- Change of studio name on StashDB (for canonicalisation reasons)
- Change of studio name on StashDB, because it's actually changed it's name (SinsVR => XsinsVR) 

There are both example of slight changes of names and many whitespace changes. The studio aliases are supported in stash app, so if a query is done against stashDB and good studio aliases are returned when a studio is matched, it will match the canonical studio even if the alias is scraped

**Comment by feederbox826** [comment]
The canonical and "Official" names seem to also differ and can be volatile or in some cases extremely inconsistent. https://github.com/stashapp/CommunityScrapers/pull/2028 documents such a change in KellyMadison

And my own deep dive into BaDoink naming schemes frankly annoyed me since `customized-brand-name` is used in the communityScrapers because of it's consistency

| sDB | application-name | customization-brand-name | og:title | description | search |
|---|---|---|---|---|---|
| 18VR | 18vr | 18 VR | 18VR | 18VR | 18VR |
| 18Yoga |  |  | 18Yoga.com |  |  |
| BabeVR |  | Babe VR | BabeVR |  | BabeVR |
| BaDoinkVR | Badoink VR | BaDoink VR | BaDoinkVR | BaDoinkVR | BaDoinkVR |
| Banned Stories |  |  | bannedstories | BannedStories |  |
| KinkVR |  | KinkVR | KinkVR | KinkVR | KinkVR |
| RealVR |  | RealVR | RealVR |  | RealVR |
| VRCosplayX | VRCosplayX | VR CosplayX | VRCosplayX | VRCosplayX | VRCosplayX |

[discord](https://discord.com/channels/559159668438728723/1285023577099472937)

**Comment by sleetx** [comment]
Studio start and end date is something I've fleshed out more in a separate feature request #820 . Although maybe it could be rolled into this. I think those fields need to be expanded on to give a complete picture of their release + ownership timeline.

**Comment by echo6ix** [comment]
The same thing is happening with expanded studio fields that happened with performer and scene fields. Over time people begin thinking of good ideas for pragmatic fields. Or fields that are specific to StashDB, but may be unnecessary to other stash-boxes or vice versa.

I'd propose making a system where the stash-box administrator can add custom fields to objects (studios), rather than hard-coding them. After all, stash-box is more generalized than StashDB, as such offering custom fields makes it more adaptable.
